substrate-node/support/src/resources.rs

```diff
@@ -45,10 +57,38 @@ impl Resources {
         self.mru as u128 >= 2 * GIGABYTE
     }
 
+    /// Compute the CU, SU and NU for the node. The result is expressed in a "permill" way. So the
+    /// actual CU, SU and NU are obtained by dividing the results by 1_000_000.
+    ///
+    /// In order for this to be accurate, the data about network and IP usage needs to already have
+    /// been aggregated on the node object.
+    ///
+    /// Calculation taken from [the
+    /// wiki](https://library.threefold.me/info/threefold#/tfgrid/farming/threefold__resource_units_calc_cloudunits)
+    /// on 31-01-2022 as follows:
+    ///   CU: MIN(cru * 4 / 2, (mru - 1) / 4, sru / 50)
+    ///   SU: hru / 1200 + sru * 0.8 / 200
+    pub fn cloud_units_permill(&self) -> (u64, u64) {
+        // Calculate CU first. Mru and sru are in bytes, but are expressed in GB in the formula.
+        // Rather than dividing first, we multiply cru first, then take the MIN, and finally
```

Comment:
Maybe get more detailed comment here because not easy to understand precision trick
~~~rust
// Rather than converting mru and sru to GB (by dividing first), we multiply cru first instead (to put the values at same scale), then take the MIN, and finally divide the returning values at the end of function
~~~